src/main/scala/org/renci/cam/LookupService.scala

```diff
@@ -0,0 +1,420 @@
+    *   The relations connected to the object of this relation. CAM-KP is designed so that this list is not necessary, since `subj` should
+    *   be connected to every other entity in its graph (even if only by a `biolink:related_to` predicate); however, it is intended to test
+    *   whether this is in fact the case. TODO: not yet implemented.
+    * @param g
+    *   The graph that this relation comes from.
+    */
+  case class Relation(
+    subj: Set[LabeledIRI],
+    preds: Set[LabeledIRI],
+    biolinkPredicates: Map[String, Set[LabeledIRI]],
+    obj: Set[LabeledIRI],
+    objRelations: Seq[Relation],
+    g: Set[String]
+  )
+
+  /** The result of calling `/lookup` on an identifier.
+    *
+    * @param queryId
+    *   The identifier queried with lookup.
+    * @param normalizedIds
+    *   The list of normalized identifiers for the queryId with their labels. If the node could not be normalized, this will be a single
+    *   entry, which consists of queryId without any labels. All of these identifiers will be used in the query.
+    * @param biolinkPredicates
+    *   All predicates connected to the identifier being looked up.
+    * @param relations
+    *   All relations connected to the identifier being looked up.
+    * @param subjectTriples
+    *   All subject triples for the identifier being looked up (i.e. triples in the triplestore in the shape `normalizedId ?p ?o`).
+    * @param objectTriples
+    *   All object triples for the identifier being looked up (i.e. triples in the triplestore in the shape `?s ?p normalizedId`)
+    */
+  case class Result(
+    queryId: String,
+    normalizedIds: List[LabeledIRI],
+    biolinkPredicates: Map[String, Set[LabeledIRI]],
+    relations: Seq[Relation],
+    subjectTriples: Seq[ResultTriple],
+    objectTriples: Seq[ResultTriple]
+  )
+
+  /** An error returned by calling `/lookup`. */
+  case class Error(
+    code: String,
+    message: String
+  )
+
+  /* CONTROLLER */
+  type LookupEndpointParams = (String, Int, Option[String], String)
+  type LookupEndpoint = Endpoint[LookupEndpointParams, Error, Result, Any]
+
+  /* Helper methods */
+
+  /** Returns a list of Relations for a particular subject.
+    *
+    * TODO: in a future version, this will call itself recursively so we can fill in objRelations.
+    *
+    * @param qualifiedIds
+    *   The subject IRIs to find relations for.
+    */
+  def getRelations(
+    qualifiedIds: Set[String]): ZIO[ZConfig[AppConfig] with HttpClient with Has[BiolinkData], Throwable, Iterable[Relation]] =
+    for {
+      biolinkData <- biolinkData
+
+      subjectIRIs = {
+        implicit val iriKeyDecoder: KeyDecoder[IRI] = Implicits.iriKeyDecoder(biolinkData.prefixes)
+
+        qualifiedIds.map(qualifiedId => iriKeyDecoder(qualifiedId).get)
+      }
+
+      relationsQuery =
+        sparql"""PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+               PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
```

Comment:
Don't use prefixes in sparql interpolator queries. Just embed the constant IRI object where needed.

Reply:
Fixed in a664bdc.